python/hail/utils/misc.py

```diff
@@ -310,17 +310,15 @@ def check_keys(name, indices):
         raise ExpressionException(msg)
 
 def get_select_exprs(caller, exprs, named_exprs, indices, protect_keys=True):
-    from hail.expr.expressions import to_expr, ExpressionException
+    from hail.expr.expressions import to_expr, ExpressionException, analyze
     exprs = [to_expr(e) if not isinstance(e, str) else indices.source[e] for e in exprs]
     named_exprs = {k: to_expr(v) for k, v in named_exprs.items()}
     assignments = OrderedDict()
 
     for e in exprs:
-        if not e._indices == indices:
-            raise ExpressionException("method '{}' parameter 'exprs' expects {}-indexed fields,"
-                                  " found indices {}".format(caller, list(indices.axes), list(e._indices.axes)))
         if not e._ir.is_nested_field:
             raise ExpressionException("method '{}' expects keyword arguments for complex expressions".format(caller))
+        analyze(caller, e, indices, broadcast=False)
```

Comment:
I think this needs to come before the above if block, in order to correct #4093.

Reply:
I think I want the above error if somebody does `select_rows(ht.x + 2)`. Lines 319-321 were the ones that threw the out-of-place error here (indices instead of source). I tested the specific example locally and the error is what we want.